### PR TITLE
enhance COCODataset

### PR DESCRIPTION
**Pull request by Menooker**
## What changes were proposed in this pull request?

Add COCODataset.loadFromSeqFile - load from seq file
Add COCODataset.toImageFeatures - convert local file COCO into Iterator[ImageFeature]
Add COCOImage.toTable

## How was this patch tested?

UT

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala [hunk 1]
```diff
@@ -597,14 +597,37 @@ object DataSet {
       if (image.getRaster.getNumBands == 1) return true
       false
     }
+
+    /**
+     * Decode raw bytes read from an image file into decoded bytes. If the file is 1 channel grey
+     * scale image, automatically convert to 3 channels
+     * @param in the input raw image bytes
+     * @return the decoded 3 channel bytes
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala]: BGR format

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala [hunk 2]
```diff
@@ -597,14 +597,37 @@ object DataSet {
...
+     */
+    private[bigdl] def decodedRawImageFileBytes(in: Array[Byte]) : Array[Byte] = {
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala]: decodedRawImageToBGR

**Comment by zhangxiaoli73** [comment]
LGTM

**Comment by Menooker** [comment]
Jenkins pass http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/1028/

**Pull request merged**